src/components/header/header.js

```diff
@@ -23,15 +24,40 @@ export default class Header {
       }
       header = createElementsFromString(usersHeader(headersProps));
     }
-    const screenWithHeader = target.appendChild(header);
+    if (target.firstChild) {
+      replaceFirstChild(target, header);
+    } else {
+      target.appendChild(header);
+    }
     if (type === roles.manager) {
-      target.querySelector('.header-content__switch-mode-button').addEventListener('click', () => {
+      header.querySelector('.header-content__switch-mode-button').addEventListener('click', () => {
         switchMode(props);
       });
     }
-    target.querySelector('.exit-ico').addEventListener('click', () => { logout(props.router); });
+    header.querySelector('.exit-ico').addEventListener('click', () => {
+      logout(props.router);
+    });
+
+    // to remove
+    if (props.page === 'user') {
+      header.querySelector('.history-ico').addEventListener('click', () => {
+        eventBus.publish('onBalanceChange', 260);
+      });
+    }
+    //
+
+    this.unsubscribe = eventBus.subscribe('onBalanceChange', this.onBalanceChange.bind(this, target, props));
+
+    return header;
+  }
+
+  onBalanceChange(target, props, balance) {
+    setCookie('balance', balance, 365);
```

Comment:
I guess it is not the right place to update cookie. So, in general Header depends on cookie and the right event change should looks like 
BalanceUpdate -publish balance update-> CookieUpdate -publish cookie update-> Component react on cookie update and do not react on BalanceUpdate